### ISSUE TITLE
Kill exec process with SIGTERM instead of SIGKILL

### ISSUE THREAD

**Issue by inoc603** [containerd/go-runc#21]
In `runc.Exec`, [here](https://github.com/containerd/go-runc/blob/master/runc.go#L201) the exec command is created with `exec.CommandContext`, which will kill the `runc` command with `os.Process.Kill` when the context is done. And when `runc` is killed by `SIGKILL`, the exec process inside the container is not stopped. I think we should send `SIGTERM` manually to stop the `runc` command when we're doing exec.

I only tested this with `docker-runc`. Can anyone confirm whether it is the same with `runc`?

**Comment by crosbymichael** [comment]
Why do you think this is not the correct functionality?  When a context closes, that means the request or parent crashed/ended so we don't want to orphan the runc process.

**Comment by inoc603** [comment]
Yes the runc process should be killed when the context closes. My point is that it should be killed by `SIGTERM` rather than `SIGKILL`, so runc itself can clean up the exec process in the container.  Current mechanism will orphan the exec process inside the container.

What I'm suggesting is listening for `ctx.Done()` manually, instead of letting the `exec` package handle it.

**Comment by crosbymichael** [comment]
@inoc603 ok, the hard part is that Go is the one that sends the `SIGKILL`.  We would have to rewrite all the logic for `CommandContext` to make this work.

**Comment by dmcgowan** [comment]
Closed #28 for staleness, is this issue still relevant though?
